uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/NavigationMenuHelper.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+/**
+ * Helper class to interact with a CiviCRM navigation menu array
+ */
+class CRM_HRCore_Helper_NavigationMenuHelper {
+
+  /**
+   * Recursively search for a menu item by its name
+   *
+   * @param array $menu
+   *   The menu array to search in
+   * @param string $path
+   *   The path to the item, e.g. "Events/Dashboard"
+   *
+   * @return array|null
+   *   The menu item, if found, or null otherwise
+   */
+  public static function findMenuItemByPath($menu, $path) {
+    return self::findMenuItemReferenceByPath($menu, $path);
+  }
+
+  /**
+   * Inserts a navigation menu item as a child of the given parent
+   *
+   * @param array $menu
+   *   menu hierarchy
+   * @param string $parentPath
+   *   Path to the parent menu item, e.g. "Events/Dashboard"
+   * @param array $itemAttributes
+   *   Attributes of item to insert, e.g. ['name' => 'New Item']
+   */
+  public static function insertChild(&$menu, $parentPath, $itemAttributes) {
+    $parent = &self::findMenuItemReferenceByPath($menu, $parentPath);
+
+    if (!$parent) {
+      $err = sprintf('Cannot find parent item "%s"', $parentPath);
+      throw new \Exception($err);
+    }
+
+    $label = CRM_Utils_Array::value('name', $itemAttributes);
+    $defaults = ['label' => $label, 'active' => 1];
+    $newItem = ['attributes' => array_merge($defaults, $itemAttributes)];
+
+    $parent['child'][] = $newItem;
+  }
+
+  /**
+   * Moves a menu item to after another specified menu item
+   *
+   * @param array $menu
+   *   The full menu structure
+   * @param string $itemToMovePath
+   *   The path to the menu item we want to move
+   * @param string $precedingItemPath
+   *   The path to the item that the target should be inserted after
+   */
+  public static function relocateAfter(
+    &$menu,
+    $itemToMovePath,
+    $precedingItemPath
+  ) {
+    $siblings = &self::getSiblingsReference($menu, $precedingItemPath);
+    $itemToMove = self::findMenuItemByPath($menu, $itemToMovePath);
+
+    // remove original from the menu
+    self::remove($menu, $itemToMovePath);
+
+    // re-index and find point to insert the new item
+    $siblings = array_values($siblings);
```

Comment:
I hope there are no consequneces for re-indexing the array?

Reply:
I hope so too, originally they are indexed by ID. As part of this feature we will have new menu items too, which will have no ID, so there would be problems there too. However, from testing it locally I found no issues.